src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceDao.java

```diff
@@ -255,6 +260,30 @@ private GoogleProjectResource retrieveProjectBy(String sql, MapSqlParameterSourc
                 .googleProjectNumber(rs.getString("google_project_number")));
   }
 
+  @Transactional(
+      propagation = Propagation.REQUIRED,
+      isolation = Isolation.SERIALIZABLE,
+      readOnly = true)
+  public Integer countSnapshotsUsingProject(UUID googleProjectId) {
```

Comment:
One overall comment, it looks like this code should use `int` not `Integer` for these results, as the code eventually converts them to `int`. so it's assuming that they are non-null.

In some cases we have code that checks for a null result from SQL `COUNT` and throws `CorruptMetadataException` which you could do if you like. IMO letting the code throw a NPE is fine in this case. (See `DataStorageAccountDao.datasetStorageAccountLinkExists()`)

Reply:
I disagree here (not with having this return an `int` but with the NPE)...NPEs should be avoided.  Especially when it's from trying to cast to a primitive.  Those are notorious red herrings and make you lose time in debugging and provide not very useful error messages I find